LoopKitUI/Views/TherapySettingsView.swift

```diff
@@ -0,0 +1,342 @@
+                    workoutTargetRange: self.therapySettings.workoutTargetRange,
+                    unit: self.glucoseUnit,
+                    preset: preset,
+                    correctionRangeScheduleRange: self.therapySettings.glucoseTargetRangeSchedule?.scheduleRange()
+                )
+            }
+        }
+    }
+    
+}
+
+struct ScheduleRangeItem: View {
+    let time: TimeInterval
+    let range: DoubleRange
+    let unit: HKUnit
+    let guardrail: HKQuantityGuardrail
+    
+    public var body: some View {
+        ScheduleItemView(time: time,
+                         isEditing: .constant(false),
+                         valueContent: {
+                            GuardrailConstrainedQuantityRangeView(range: range.quantityRange(for: unit), unit: unit, guardrail: guardrail, isEditing: false)
+                         },
+                         expandedContent: { EmptyView() })
+    }
+}
+
+struct CorrectionRangeOverridesRangeItem: View {
+    let preMealTargetRange: DoubleRange?
+    let workoutTargetRange: DoubleRange?
+    let unit: HKUnit?
+    let preset: CorrectionRangeOverrides.Preset
+    let correctionRangeScheduleRange: ClosedRange<HKQuantity>?
+    
+    public var body: some View {
+        CorrectionRangeOverridesExpandableSetting(
+            isEditing: .constant(false),
+            value: .constant(CorrectionRangeOverrides(
+                preMeal: preMealTargetRange,
+                workout: workoutTargetRange,
+                unit: unit
+            )),
+            preset: preset,
+            unit: unit,
+            correctionRangeScheduleRange: correctionRangeScheduleRange,
+            expandedContent: { EmptyView() })
+    }
+}
+
+struct SectionHeaderWithEdit: View {
+    @Binding var isEditing: Bool
+    @Binding var editButtonDesignVersion: TherapySettingsView.DesignVersion
+    let title: String
+    let editAction: () -> Void
+
+    public var body: some View {
+        HStack(alignment: .firstTextBaseline) {
+            SectionHeader(label: title)
+            if isEditing && editButtonDesignVersion == .C {
+                Spacer()
+                Button(action: editAction) {
+                    Text("Edit")
+                        .font(.subheadline)
+
+                }
+                .disabled(!isEditing)
+            }
+        }
+    }
+}
+
+// Note: I didn't call this "EditableSection" because it doesn't actually make the section editable,
+// it just optionally provides a link to go to an editor screen.
+struct SectionWithEdit<Content, Footer>: View where Content: View, Footer: View {
+    @Binding var isEditing: Bool
+    @Binding var editButtonDesignVersion: TherapySettingsView.DesignVersion
+    let title: String
+    let footer: Footer
+    let editAction: () -> Void
+    let content: () -> Content
+
+    public var body: some View {
+        buildBody()
+    }
+    
+    @ViewBuilder private func buildBody() -> some View {
```

Comment:
You can just annotate `body` with `@ViewBuilder` and move the body of this function to `body`. The annotation works on computed properties.

Reply:
Huh.  I could've sworn I tried that and it didn't compile, but it is working now.  Thanks!